sdk/core/Azure.Core/src/Shared/NextLinkOperationImplementation.cs

```diff
@@ -152,16 +173,20 @@ internal class NextLinkOperationImplementation : IOperation
             _finalStateVia = finalStateVia;
             _pipeline = pipeline;
             _apiVersion = apiVersion;
-            OperationId = ParseOperationId(nextRequestUri);
+            OperationId = ParseOperationId(startRequestUri, nextRequestUri);
         }
 
-        private static string? ParseOperationId(string nextRequestUri)
+        private static string ParseOperationId(Uri startRequestUri, string nextRequestUri)
         {
-            if (Uri.TryCreate(nextRequestUri, UriKind.Absolute, out var uri))
+            if (Uri.TryCreate(nextRequestUri, UriKind.Absolute, out var nextLink) && nextLink.Scheme != "file")
```

Comment:
It would be helpful to add a comment explaining why we do something special for `file`.

I'm also not entirely sure the logic is correct here -- if TryCreate returns `false`, do you know that the `new Uri` call on line 187 will succeed?

Also, you are returning from both the `if` block and the `else` block -- I think this means line 189 can never be reached?  Is this what you intended?  If so, what does line 189 accomplish?

Reply:
I borrowed the logic from existing code [here](https://github.com/Azure/azure-sdk-for-net/blob/c3105fa3c4fa4071918def584f010fc1cd37e7bc/sdk/core/Azure.Core/src/Shared/NextLinkOperationImplementation.cs#L286), there are cased nextLinkUri is incomplete(host missing).
You are right, the exception thrown should be removed.